scripts/add_electricity.py

```diff
@@ -338,6 +348,24 @@ def attach_wind_and_solar(
                 p_max_pu=ds["profile"].transpose("time", "bus").to_pandas(),
             )
 
+        if suptech == "offwind":
+            continue
+            # df.carrier.mask(df.technology=="Offshore", "offwind-ac",inplace=True)
+
+        df.carrier.mask(df.technology == "Onshore", "onwind", inplace=True)
+
+        gens = n.generators[lambda df: df.carrier == tech]
```

Comment:
Instead of the lambda function could you try this:
`n.generators[df.carrier == tech]`

Reply:
@pz-max, if we put the code in front to make `madd()` instead of `update`, it need to be rewritten majorly. I will do that. Regarding not using lambda function, if we omit lambda function, it will raise error: "Unalignable boolean Series provided as indexer (index of the boolean Series and of the indexed object do not match)." I will figure everything out.